### PR TITLE
Add support to indicate synthetic exception handler

### PR DESCRIPTION
**Pull request by cathyzhyi**
Sometimes compiler can choose to insert exception handlers that do not
exit in the source code. Special handling of those handler
might be necessary. In case of OSR, the runtime would need to
handle synthetic handler differently than a normal handler.

This PR adds support to indicate a certain block or handler is for
a synthetic exception handler. It also adds the special helper for throwing
exceptions from the synthetic handler.

see discussion in : https://github.com/eclipse/openj9/issues/9267

Signed-off-by: Yi Zhang <yizhang@ca.ibm.com>

### CODE REVIEW

**Comment by andrewcraik** [comment]
This seems reasonable to me, but I would like feedback from @vijaysun-omr as well since he had some differing opinions in the OpenJ9 discussion and I would like to make sure this implementation is consistent with his concerns.

File: compiler/runtime/Helpers.inc [hunk 1]
```diff
@@ -50,7 +50,8 @@ SETVAL(TR_writeBarrierBatchStore,TR_writeBarrierStoreRealTimeGC+1)
 SETVAL(TR_arrayStoreException,TR_writeBarrierBatchStore+1)
 SETVAL(TR_throwClassCastException,TR_arrayStoreException+1)
 SETVAL(TR_throwCurrentException,TR_throwClassCastException+1)
-SETVAL(TR_IncompatibleClassChangeError,TR_throwCurrentException+1)
+SETVAL(TR_throwUnreportedException,TR_throwCurrentException + 1)
```
Comment (vijaysun-omr) [inline compiler/runtime/Helpers.inc:53]: What does this line have to do with the rest of the commit that seems to be more about marking and checking a synthetic exception handler ?

**Comment by vijaysun-omr** [comment]
Thanks @andrewcraik for the ping. I feel this commit does'nt have any of the issues that led to me expressing concerns in the downstream project discussion. So I am fine with it in principle, modulo the question I asked in the code review.

File: compiler/runtime/Helpers.inc (hunk 1 above)
Comment (cathyzhyi) [reply compiler/runtime/Helpers.inc:53]: It's for adding the VM helper for throwing exceptions from the synthetic exception handler. It's also part of eclipse/openj9#9267.

File: compiler/runtime/Helpers.inc [hunk 2]
```diff
@@ -50,7 +50,8 @@ SETVAL(TR_writeBarrierBatchStore,TR_writeBarrierStoreRealTimeGC+1)
...
+SETVAL(TR_IncompatibleClassChangeError,TR_throwCurrentException+2)
```
Comment (gacholio) [inline compiler/runtime/Helpers.inc:54]: Should this be `TR_throwUnreportedException+1` ?

Comment (cathyzhyi) [reply compiler/runtime/Helpers.inc:54]: `TR_throwCurrentException+2` and `TR_throwUnreportedException+1` are actually equal.

Comment (gacholio) [reply compiler/runtime/Helpers.inc:54]: Yes, but it seems an unnecessary change in the pattern.

Comment (cathyzhyi) [reply compiler/runtime/Helpers.inc]: changed to `TR_throwUnreportedException+1` as suggested.

File: compiler/runtime/Helpers.inc (hunk 1 above)
Comment (andrewcraik) [reply compiler/runtime/Helpers.inc]: @vijaysun-omr can you resolve the discussion if this answer is acceptable?

**Comment by cathyzhyi** [comment]
Fixed a few places to properly initialize`_isSyntheticHandler` field.

File: compiler/runtime/Helpers.inc (hunk 1 above)
Comment (vijaysun-omr) [reply compiler/runtime/Helpers.inc]: Yes I was just wondering why this change is there when it is not mentioned in the commit or PR message as such. I obviously have no issue with adding the helper.

**Comment by andrewcraik** [comment]
@genie-omr build all

File: compiler/runtime/Helpers.inc (hunk 1 above)
Comment (cathyzhyi) [reply compiler/runtime/Helpers.inc]: Added description for the helper int the PR description.

**Comment by andrewcraik** [comment]
All concerns appear addressed and reviews complete - merging

**Pull request merged**